### PR TITLE
chore: Replace login client with resources client

### PR DESCRIPTION
**Pull request by dbolson**
Replaces an interface and struct type with an embedded interface on an existing one. We can remove a bunch of almost-duplicate code.
**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: internal/resources/client.go [hunk 1]
```diff
@@ -10,7 +11,16 @@ import (
 	"github.com/launchdarkly/ldcli/internal/errors"
 )
 
+type UnauthenticatedClient interface {
```
Comment (dbolson) [inline internal/resources/client.go:14]: We can use this interface in the login tests so we don't need to implement an unused `MakeRequest` method.

File: internal/resources/client.go [hunk 2]
```diff
@@ -24,9 +34,21 @@ func NewClient(cliVersion string) ResourcesClient {
 	return ResourcesClient{cliVersion: cliVersion}
 }
 
-func (c ResourcesClient) MakeRequest(accessToken, method, path, contentType string, query url.Values, data []byte, isBeta bool) ([]byte, error) {
-	client := http.Client{}
+func (c ResourcesClient) MakeUnauthenticatedRequest(
```
Comment (dbolson) [inline internal/resources/client.go:37]: Now both clients share the same code, so we don't need to worry about drifting implementations.

File: internal/resources/client.go [hunk 3]
```diff
@@ -47,9 +69,22 @@ func (c ResourcesClient) MakeRequest(accessToken, method, path, contentType stri
...
+	if res.StatusCode < http.StatusBadRequest {
+		return body, nil
+	}
+
+	if len(body) > 0 {
 		return body, errors.NewError(string(body))
 	}
 
-	return body, nil
+	switch res.StatusCode {
```
Comment (dbolson) [inline internal/resources/client.go:80]: These changes were pulled in from the unauthenticated client.

**Review by k3llymariee — APPROVED** [review]
oh neat - we can use this over in the launchdevly stuff too!

**Pull request merged**